starlite/app.py

```diff
@@ -207,6 +206,7 @@ def __init__(
         response_headers: OptionalSequence[ResponseHeader] = None,
         security: OptionalSequence[SecurityRequirement] = None,
         static_files_config: OptionalSequence[StaticFilesConfig] = None,
+        stores: StoreRegistry | dict[str, Store] | None = None,
```

Comment:
I think this API is bad. 

1. It's not ergonomic at all. 
2. It's overly complex.
3. It's not junior friendly. 

Why not simply let the user specify either a single or a sequence if store configs and pass them to the app? Expose an `is_default_store` on the config, and also a `key`, and you can easily generate a mapping from this internally. 

You also allow plugins yo modify it, as needed.

Reply:
How is `[StoreConfig(store=MemoryStore(), key="memory"), StoreConfig(store=RedisStore(), key="redis")]` more ergonomic than `{"memory": MemoryStore(), "redis": RedisStore()}`?